.planning/phases/01-sdk-core-implementation/01-RESEARCH.md

```
@@ -0,0 +1,643 @@
+|------------|-----------|----------|
+| vitest bench | benchmark.js (2.1.4) | benchmark.js is more mature but requires separate dependency; vitest bench is built-in, consistent with existing test infrastructure, produces p99 out of the box. Per D-04, vitest bench is locked. |
+| Separate @principles/adapters package | All in @principles/core | Separate packages allow independent versioning but add workspace complexity. Per D-02, single package is locked. |
+
+**Installation:**
+```bash
+# No new packages needed -- all dependencies already in monorepo
+# The new packages/principles-core/ reuses existing workspace dependencies
+```
+
+**Version verification:**
+```
```

markdownlint-cli2 (0.22.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)